### DISCUSSION TITLE
How can I direct something.html -> something?

### DISCUSSION THREAD

**Discussion by cellio** [datenstrom/community#687] [Ask a question] [answered]
Pages I create in Yellow have URLs of the form "example.com/something", where "something" is generated from something.md.  I want to replace my current (hand-built, ugly) web site with one run from Yellow.  I have existing (published) URLs of the form "example.com/something.html" and I need to make those work.  There is a finite number of those, so I could do individual redirects, but I can't seem to get that to work.  Is there a way to configure Yellow to put the .html extension on the URLs it generates?

What I tried: I tried adding the following to .htaccess (based on something I found in a web search):

    RewriteRule ^(.+)\.html$ $1 [L,NC,R=301] 

But when I visit something.html, I get a "file not found" page (presumably a 404).

I then tried pattern-matching the last line in the .htaccess config that the Yellow troubleshooting doc suggests, and came up with this (including the full block for context):

    <IfModule mod_rewrite.c>
    RewriteEngine on

    DirectoryIndex index.html yellow.php
    RewriteRule ^(cache|content|system)/ error [L]
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteRule ^(.*)\.html$ yellow.php/$1 [L]
    RewriteRule ^ yellow.php [L]
</IfModule>

The penultimate RewriteRule is the one I added.  I assume these rules are processed in order and that I needed to put it before the final one.  I also don't assume I know what I'm doing with regex.  Or maybe the regex is fine *but* that's not how to invoke Yellow on a specific page.

I looked through `yellow-system.ini` for things that looked promising but found only `CoreServerUrl`, which doesn't seem to be what I want.  I looked at the documentation for the Core extension.

I would be happy with any of:

- All URLs showing as .html rather than just the slug (is slug the right term?)
- File settings in the Markdown headers for the URLs I want to modify (e.g. I set some value in something.md that says "make this one something.html instead of, or in addition to, just something")
- Web-server redirects for the published URLs I care about
- Some other way to not break those URLs that I haven't thought of

Thanks for any help y'all can offer!  I really appreciate the help I got with my preview newbie question.

**Comment by nzyeehaa** [comment, marked as answer]
Hi @cellio 
first - when testing rewrites - dont use 301 until it is all working as the browser will remember and block them 
second - each host and setup is different / as is development vs live - so you may need some changes
third - be aware of the way yellow URL's work _/test/page.md_ (which is the default index page in test folder) will show as **/test/** _but_  /test/nextpage.md will show as **/test/nextpage** (with no slash)

then - modify .htaccess with redirects
```
RewriteEngine on
Redirect /old-url.html /old-url/
Redirect /test.html /test/
Redirect /test/nextpage.html /test/nextpage
DirectoryIndex index.html yellow.php
```

when all old pages are mapped correctly, working properly, and testing is complete - update redirects to permanent...
`Redirect 301 /old-url.html /old-url`

And some years later, when search results are correct and people have updated their bookmarks, delete the redirects

**Reply by cellio** [reply]
Thanks!  Redirects, not rewrites -- that and the '/' were the clues I was missing.

**Reply by markseuffert** [reply]
Thanks! Added another example to datenstrom/community#138.

**Reply by cellio** [reply]
Thanks -- I didn't find that issue when I searched before asking (I guess because I focused too much on the .html part, not realizing it was a more general question).
